src/main/java/seedu/address/logic/commands/DeleteTagCommand.java

```diff
@@ -31,77 +28,107 @@ public class DeleteTagCommand extends UndoableCommand {
      */
-    public DeleteTagCommand(List<Tag> tags) {
+    public DeleteTagCommand(List<Tag> tags, boolean isForceDelete) {
         requireAllNonNull(tags);
         this.tags = tags;
+        this.isForceDelete = isForceDelete;
     }
 
-    /**
-     * Removes the specified tags from all persons who are tagged with these tags.
-     *
-     * @param model Model that contains the list of persons to remove tags from.
-     */
-    private void removeTagsFromPersons(Model model) {
-        for (Tag tag : tags) {
-            deletedSet.put(tag, model.removeTagFromPersons(tag));
-        }
-    }
-
-    /**
-     * Creates and returns a CommandResult based on the success of the deletion.
-     *
-     * @param isSuccessful Boolean indicating if the deletion was successful.
-     * @return CommandResult indicating the outcome of the deletion attempt.
-     */
-    private CommandResult createCommandResult(boolean isSuccessful) {
-        if (!isSuccessful) {
-            return new CommandResult(MESSAGE_NONEXISTENT);
-        }
-        return new CommandResult(MESSAGE_SUCCESS);
-    }
-
-    /**
-     * Executes the DeleteTagCommand by deleting the specified tags from the tag list in the model.
-     * If the tags are currently being used to tag any persons, the user is prompted for confirmation.
-     *
-     * @param model Model containing the tags and persons to operate on.
-     * @return CommandResult indicating the result of the command.
-     * @throws CommandException If the user cancels the deletion confirmation.
-     */
     @Override
     public CommandResult execute(Model model) throws CommandException {
         requireAllNonNull(model);
+        StringBuilder errorMsg = new StringBuilder();
+        StringBuilder successMsg = new StringBuilder();
+        boolean showErrorMsg = false;
+        boolean showSuccessMsg = false;
+
+        // Proceed with force deletion logic
+        if (isForceDelete) {
+            for (Tag tag : tags) {
+                if (!model.hasTag(tag)) {
+                    errorMsg.append(tag.toString() + " ");
+                    showErrorMsg = true;
+                }
+            }
 
-        // Get a set of tags currently being used to tag any Person.
-        Set<Tag> tagsInUse = model.getTagsInUse();
+            for (Tag tag : tags) {
+                if (model.hasTag(tag)) {
+                    model.deleteTags(List.of(tag)); // Remove from model tags
+                    model.removeTagFromPersons(tag); // Remove from all persons
+                    showSuccessMsg = true;
+                }
+            }
 
-        if (!tagsInUse.isEmpty()) {
-            getUserConfirmation(tagsInUse);
-        }
+            successMsg.append(MESSAGE_SUCCESS_FORCE);
+            String successMessage = successMsg.toString();
 
-        boolean isSuccessful = model.deleteTags(tags);
-        removeTagsFromPersons(model);
+            if (showErrorMsg) {
+                errorMsg.append(MESSAGE_NONEXISTENT);
```

Comment:
the createCommandResult method can be used to create success / error messages based on the status of the variables like showErrorMsg etc